doc/docs/Python_User_Interface.md

```diff
@@ -1158,7337 +1335,6462 @@ If `match_frequency=False` or `kpoint` is not zero in the given `direction`, the
+def stop_when_fields_decayed(dt, c, pt, decay_by):
 ```
 
-<div class="method_docstring" markdown="1">
+<div class="function_docstring" markdown="1">
 
+Return a `condition` function, suitable for passing to `Simulation.run` as the `until`
+or `until_after_sources` parameter, that examines the component `c` (e.g. `Ex`, etc.)
+at the point `pt` (a `Vector3`) and keeps running until its absolute value *squared*
+has decayed by at least `decay_by` from its maximum previous value. In particular, it
+keeps incrementing the run time by `dT` (in Meep units) and checks the maximum value
+over that time period &mdash; in this way, it won't be fooled just because the field
+happens to go through 0 at some instant.
 
+Note that, if you make `decay_by` very small, you may need to increase the `cutoff`
+property of your source(s), to decrease the amplitude of the small high-frequency
+components that are excited when the source turns off. High frequencies near the
+[Nyquist frequency](https://en.wikipedia.org/wiki/Nyquist_frequency) of the grid have
+slow group velocities and are absorbed poorly by [PML](Perfectly_Matched_Layer.md).
 
 </div>
 
-<a id="Simulation.output_components"></a>
-
-**Simulation.output\_components**
+<a id="stop_after_walltime"></a>
 
 ```python
-def output_components(self, fname, *components):
+def stop_after_walltime(t):
 ```
 
-<div class="method_docstring" markdown="1">
-
+<div class="function_docstring" markdown="1">
 
+Return a `condition` function, suitable for passing to `Simulation.run` as the `until`
+parameter. Stops the simulation after `t` seconds of wall time have passed.
 
 </div>
 
-<a id="Simulation.output_dft"></a>
-
-**Simulation.output\_dft**
+<a id="stop_on_interrupt"></a>
 
 ```python
-def output_dft(self, dft_fields, fname):
+def stop_on_interrupt():
 ```
 
-<div class="method_docstring" markdown="1">
+<div class="function_docstring" markdown="1">
 
-Output the Fourier-transformed fields in `dft_fields` (created by
-`add_dft_fields`) to an HDF5 file with name `fname` (does *not* include the `.h5`
-suffix).
+Return a `condition` function, suitable for passing to `Simulation.run` as the `until`
+parameter. Instead of terminating when receiving a SIGINT or SIGTERM signal from the
+system, the simulation will abort time stepping and continue executing any code that
+follows the `run` function (e.g., outputting fields).
 
 </div>
 
-<a id="Simulation.output_farfields"></a>
-
-**Simulation.output\_farfields**
 
-```python
-def output_farfields(self,
-                     near2far,
-                     fname,
-                     resolution,
-                     where=None,
-                     center=None,
-                     size=None):
-```
+Finally, another run function, useful for computing ω(**k**) band diagrams, is available via these top-level functions:
 
-<div class="method_docstring" markdown="1">
+@@ meep.run_k_points @@
+@@ meep.run_k_point @@
```

Comment:
The positioning tags for `meep_run_k_points` and `meep_run_k_point` are not being rendered correctly.

Reply:
Oops, those are actually `Simulation` methods. Fixed.